File: r/inst/examples/words_geometry.R
# words_geometry — build a PDF from Markdown, open it, extract positioned words
# from page 0 and assert the first word + its bbox. stop() (non-zero exit) on
# any assertion failure (CI smoke).
library(pdfoxide)

md <- "# Hello pdf_oxide\n\nThis is a **R** regression example.\n"

pdf <- pdf_from_markdown(md)
doc <- pdf_open_from_bytes(pdf_to_bytes(pdf))

words <- pdf_extract_words(doc, 0)
cat("word count:", length(words), "\n")

if (length(words) == 0) stop("ASSERT FAILED: no words extracted")

first <- words[[1]]
bb <- first$bbox
cat("first word:", first$text, "\n")
cat("first bbox: x=", bb$x, " y=", bb$y, " w=", bb$width, " h=", bb$height, "\n", sep = "")

if (!identical(first$text, "Hello")) {
  stop(sprintf("ASSERT FAILED: first word is '%s', expected 'Hello'", first$text))
}
if (!all(c("x", "y", "width", "height") %in% names(bb)) || !is.numeric(bb$width)) {
  stop("ASSERT FAILED: first word has no numeric bbox")
}

pdf_close(doc)
cat("WORDS OK\n")

File: dart/example/words_geometry.dart
// words_geometry — build a PDF from Markdown, open it, extract word geometry.
//
// Shared regression scenario (mirrored across language bindings). Exits
// non-zero on any failed assertion; prints "WORDS OK" on success.
import 'dart:io';

import 'package:pdf_oxide/pdf_oxide.dart';

void main() {
  final pdf = Pdf.fromMarkdown(
      '# Hello pdf_oxide\n\nThis is a **Dart** regression example.\n');
  final doc = PdfDocument.openFromBytes(pdf.toBytes());
  try {
    final words = doc.extractWords(0);
    print('word count: ${words.length}');

    if (words.isEmpty) {
      stderr.writeln('assertion failed: no words extracted');
      exit(1);
    }

    final first = words[0];
    print('first word: "${first.text}"  bbox=${first.bbox}');

    if (first.text != 'Hello') {
      stderr.writeln('assertion failed: first word is not "Hello"');
      exit(1);
    }
    if (!(first.bbox.width > 0.0 && first.bbox.height > 0.0)) {
      stderr.writeln('assertion failed: first word has no bbox');
      exit(1);
    }

    print('WORDS OK');
  } finally {
    doc.close();
    pdf.close();
  }
}
